feature_extractors.py

```diff
@@ -0,0 +1,101 @@
+## In this file few feature extraction functions are defined. Such functions don't have input arguments
```

Comment:
following on Lucas comment's, I would also add this as a class. A conceptual question, why do we flatten? I thought we would keep the N filters of the vgg as encoder_output

Reply:
Ok, I take note of the class issue. Actually I have it already done and checked. Just need to arrange the comments a little bit in the new version with a class.
With regard of the flatten, as far as I could understand, the decoder is expecting some feature vectors for each image. What we have after after the convolutional is actually N number of matrices (feature mapts) for every image. So instead of having some vectors we have some matrices. So, a flatten must be done to convert those matrices into vectors in the way that for each point in the feat maps we get a vector.  Hope it is clear? We can talk about it if somebody thinks it is not the right way but I think it matches well ith a drawing Gerard did in the jamboard.

Reply:
> following on Lucas comment's, I would also add this as a class. A conceptual question, why do we flatten? I thought we would keep the N filters of the vgg as encoder_output

By the way, notice that the flatten only flattens 2 dimensions of the tensor, the 2 and 3. So, dimension 0 are the batches, dimension 1 are the number of filters (depth) and dimensions 2 and 3 are the Height and Width of the feature maps. The dimesions that must be flatten are the last two. Hope it helps clarifying.